### PR TITLE
Add Keys Endpoints

### PR DESCRIPTION
**Pull request by colinbobolin**
### Changes

Updated SigningKeys model and added KeysClient.
See #526 for the reported issue
See [Management API Documentation](https://auth0.com/docs/api/management/v2#!/Keys/get_signing_keys)

**Disclaimer**
I need some help with the testing here. I don't want to link this up to my own Auth0 account so I was unable to run the tests locally. Please let me know if the implementation is not testing successfully.

**Endpoints**

`GET /keys/signing` - Get all application signing keys
`GET /keys/signing/{kid}` - Get an application signing key by its key id
`POST /keys/signing/rotate` - Rotate the application signing key
`PUT /keys/signing/{kid}/revoke` - Revoke an application signing key by its key id

**Classes**

`ManagementApi/Models/SigningKey` - updated model to match API documentation
`ManagementApi/Models/RevokeSigningKeyResponse` - created model to encapsulate revoked key response
`ManagementApi/Models/RotateSigningKeyResponse` - created model to encapsulate rotated key response
`ManagementApi/Clients/KeysClient` - created to encapsulate keys endpoints
`ManagementApi/ManagementApiClient` - updated model to include KeysClient
`ManagementApi.IntegrationTests/KeysTests` - created to test KeysClient

### Testing

Please describe how this can be tested by reviewers. Be specific about anything not tested and reasons why. If this library has unit and/or integration testing, tests should be added for new functionality and existing tests should complete without errors. 

- [ ] This change adds unit test coverage

- [x] This change adds integration test coverage

- [ ] This change has been tested on the latest version of the platform/language or why not

### Checklist

- [x] I have read the [Auth0 general contribution guidelines](https://github.com/auth0/open-source-template/blob/master/GENERAL-CONTRIBUTING.md)

- [x] I have read the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md)

- [ ] All existing and new tests complete without errors

### CODE REVIEW

File: src/Auth0.ManagementApi/Clients/KeysClient.cs [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+using System.Threading.Tasks;
+
+namespace Auth0.ManagementApi.Clients
+{
+    /// <summary>
+    /// Contains methods to access the /keys endpoints.
+    /// </summary>
+    public class KeysClient : BaseClient
+    {
+        readonly JsonConverter[] keysConverters = new JsonConverter[] { new PagedListConverter<SigningKey>("keys") };
```
Comment (frederikprijck) [inline src/Auth0.ManagementApi/Clients/KeysClient.cs:17]: Lets remove this converter as well as he ones in comment below.

File: src/Auth0.ManagementApi/Models/RevokeSigningKeyResponse.cs [hunk 2]
```diff
@@ -0,0 +1,19 @@
+using Newtonsoft.Json;
+
+namespace Auth0.ManagementApi.Models
+{
+    public abstract class RevokeSigningKeyResponse
+    {
+        /// <summary>
+        /// The key id of the revoked signing key
```
Comment (frederikprijck) [inline src/Auth0.ManagementApi/Models/RevokeSigningKeyResponse.cs:8]:
```suggestion
        /// The id of the revoked signing key
```

File: src/Auth0.ManagementApi/Models/RotateSigningKeyResponse.cs [hunk 3]
```diff
@@ -0,0 +1,19 @@
+using Newtonsoft.Json;
+
+namespace Auth0.ManagementApi.Models
+{
+    public abstract class RotateSigningKeyResponse
+    {
+        /// <summary>
+        /// The key id of the next signing key
```
Comment (frederikprijck) [inline src/Auth0.ManagementApi/Models/RotateSigningKeyResponse.cs:8]:
```suggestion
        /// The id of the next signing key
```

File: src/Auth0.ManagementApi/Models/SigningKey.cs [hunk 4]
```diff
@@ -1,30 +1,85 @@
...
+﻿using System;
+
+using Newtonsoft.Json;
 
 namespace Auth0.ManagementApi.Models
 {
 
     /// <summary>
-    /// 
+    /// An Application Signing Key
```
Comment (frederikprijck) [inline src/Auth0.ManagementApi/Models/SigningKey.cs:9]: I believe we can't just make changes to this class, as it is being used as part of the Client (https://github.com/auth0/auth0.net/blob/master/src/Auth0.ManagementApi/Models/Client.cs#L21) when calling https://auth0.com/docs/api/management/v2/#!/Clients/get_clients .

**Review by frederikprijck — CHANGES_REQUESTED** [review]
Thanks for the PR, Awesome work!
I left a few comments, no worries about the tests. I will verify them against my tenant and will let you know.

File: src/Auth0.ManagementApi/Models/RevokeSigningKeyResponse.cs (hunk 2 above)
Comment (frederikprijck) [reply src/Auth0.ManagementApi/Models/RevokeSigningKeyResponse.cs]: This has been resolved.

File: src/Auth0.ManagementApi/Models/RotateSigningKeyResponse.cs (hunk 3 above)
Comment (frederikprijck) [reply src/Auth0.ManagementApi/Models/RotateSigningKeyResponse.cs]: This has been resolved.

File: src/Auth0.ManagementApi/Clients/KeysClient.cs (hunk 1 above)
Comment (frederikprijck) [reply src/Auth0.ManagementApi/Clients/KeysClient.cs]: This has been resolved.

**Comment by frederikprijck** [comment]
I took a look at the tests and added a few commits, include getting the tests to work. They all run successfully, thanks!

I also resolved a few comments I made while I was at it, but there is 1 comment left. Let me know if I can be of any assistance with resolving that one.

File: src/Auth0.ManagementApi/Models/SigningKey.cs (hunk 4 above)
Comment (colinbobolin) [reply src/Auth0.ManagementApi/Models/SigningKey.cs:9]: Ah yes I absolutely missed that. Any ideas for what a new class should be called?

**Review by frederikprijck — APPROVED** [review]

**Pull request merged**